lib/services/google_analytics.rb

```diff
@@ -0,0 +1,63 @@
+require 'google/apis/analyticsreporting_v4'
+require 'googleauth'
+
+module Services
+  class GoogleAnalytics
+    include Google::Apis::AnalyticsreportingV4
+    include Google::Auth
+
+    def client(scope: "https://www.googleapis.com/auth/analytics.readonly")
+      @client ||= AnalyticsReportingService.new
+      @client.authorization ||= ServiceAccountCredentials.make_creds(
+        json_key_io: File.open(ENV["GOOGLE_AUTH_CREDENTIALS"]),
+        scope: scope
+      )
+      @client
+    end
+
+    def page_views(base_paths)
+      raise "base_paths isn't an array" unless base_paths.is_a?(Array)
+
+      request_body = build_page_views_body(base_paths)
+      data = client.batch_get_reports(request_body).reports[0].data
+
+      data.rows.inject({}) do |page_views, row|
+        page_views[row.dimensions[0]] = row.metrics[0].values[0].to_i
+        page_views
+      end
+    end
+
+    def build_page_views_body(base_paths, start_date: "7daysAgo", end_date: "today")
+      GetReportsRequest.new.tap do |reports|
```

Comment:
Would it be possible to break down this method into smaller methods? (it is going to be easier once the request object is extracted)